x/protorev/keeper/rebalance.go

```diff
@@ -0,0 +1,160 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+// IterateRoutes checks the profitability of every single route that is passed in
+// and returns the optimal route if there is one
+func (k Keeper) IterateRoutes(ctx sdk.Context, routes []gammtypes.SwapAmountInRoutes) (sdk.Coin, sdk.Int, gammtypes.SwapAmountInRoutes) {
+	var optimalRoute gammtypes.SwapAmountInRoutes
+	maxProfitInputCoin := sdk.NewCoin(types.OsmosisDenomination, sdk.ZeroInt())
+	maxProfit := sdk.ZeroInt()
+
+	for _, route := range routes {
+		// Find the max profit for the route using the token out denom of the last pool in the route as the input token denom
+		inputCoin, profit, err := k.FindMaxProfitForRoute(ctx, route, route[2].TokenOutDenom)
+		if err != nil {
+			k.Logger(ctx).Error("Error finding max profit for route: ", err)
```

Comment:
Why not return err instead of logging error in context?

Reply:
@stackman27 We wanted to both 1) be aware of any errors that occur when iterating through routes, but 2) if one route results in an error, not stop the loop of checking other routes that may be profitable.

We went with this logger / continue method to achieve the above. Do you have a better way to achieve the above / reasons why what we wanted to achieve is not ideal?